presto-native-execution/presto_cpp/presto_protocol/Base64Util.cpp

```diff
@@ -352,38 +352,53 @@ velox::VectorPtr readBlockInt(
         encoding, type, stream, pool);
   }
 
-  if (type->kind() == velox::TypeKind::ROW &&
-      isTimestampWithTimeZoneType(type)) {
-    auto positionCount = stream.read<int32_t>();
+  if (type->kind() == velox::TypeKind::ROW) {
+    if (isTimestampWithTimeZoneType(type)) {
+      auto positionCount = stream.read<int32_t>();
 
-    auto timestamps =
-        velox::BaseVector::create(velox::BIGINT(), positionCount, pool);
-    auto rawTimestamps =
-        timestamps->asFlatVector<int64_t>()->mutableRawValues();
+      auto timestamps =
+          velox::BaseVector::create(velox::BIGINT(), positionCount, pool);
+      auto rawTimestamps =
+          timestamps->asFlatVector<int64_t>()->mutableRawValues();
 
-    auto timezones =
-        velox::BaseVector::create(velox::SMALLINT(), positionCount, pool);
-    auto rawTimezones = timezones->asFlatVector<int16_t>()->mutableRawValues();
+      auto timezones =
+          velox::BaseVector::create(velox::SMALLINT(), positionCount, pool);
+      auto rawTimezones =
+          timezones->asFlatVector<int16_t>()->mutableRawValues();
 
-    velox::BufferPtr nulls = readNulls(positionCount, stream, pool);
-    const uint64_t* rawNulls =
-        nulls == nullptr ? nullptr : nulls->as<uint64_t>();
+      velox::BufferPtr nulls = readNulls(positionCount, stream, pool);
+      const uint64_t* rawNulls =
+          nulls == nullptr ? nullptr : nulls->as<uint64_t>();
 
-    for (auto i = 0; i < positionCount; i++) {
-      if (!rawNulls || !velox::bits::isBitNull(rawNulls, i)) {
-        int64_t unpacked = stream.read<int64_t>();
-        unpackTimestampWithTimeZone(
-            unpacked, rawTimestamps[i], rawTimezones[i]);
+      for (auto i = 0; i < positionCount; i++) {
+        if (!rawNulls || !velox::bits::isBitNull(rawNulls, i)) {
+          int64_t unpacked = stream.read<int64_t>();
+          unpackTimestampWithTimeZone(
+              unpacked, rawTimestamps[i], rawTimezones[i]);
+        }
       }
+
+      return std::make_shared<velox::RowVector>(
+          pool,
+          velox::TIMESTAMP_WITH_TIME_ZONE(),
+          nulls,
+          positionCount,
+          std::vector<velox::VectorPtr>{timestamps, timezones});
     }
 
+    auto numFields = stream.read<int32_t>();
```

Comment:
Let's remove this logic outside to reduce the number of nested condition check 
Right now it's 
```
if() {
   if () {
   }
}
```

We can change it to
```
if() {
}
else if () {
}
```